docs/robotframework-tutorial.rst

```diff
@@ -0,0 +1,382 @@
+following into this file, and then save it.
+
+.. code-block:: console
+
+    *** Settings ***
+    Resource  cumulusci/robotframework/Salesforce.robot
+    Library   cumulusci.robotframework.PageObjects
+
+    Suite Setup     Open test browser
+    Suite Teardown  Delete records and close browser
+
+    *** Test Cases ***
+    Test the MyObject listing page
+        Go to page  Listing  MyObject__c
+        Current page should be  Listing  MyObject__c
+
+.. note::
+
+   The above code uses ``Go to page`` and ``Current page should be``
+   which accept a page type (Listing) and object name
+   (MyObject__c). Even though we have yet to create that page object,
+   the keywords will work by using a generic implementation. Later,
+   once we've created the page object, the test will start using our
+   implementation.
+
+To run just this test, run the following command at the prompt:
+
+.. code-block:: console
+
+    $ cci task run robot -o suites robot/MyProject/tests/MyObject.robot --org dev
+
+If everything is set up correctly, you should see the output that
+looks similar to this:
+
+.. code-block:: console
+
+    $ cci task run robot -o suites robot/MyProject/tests/MyObject.robot --org dev
+    2019-05-21 17:29:25: Getting scratch org info from Salesforce DX
+    2019-05-21 17:29:29: Beginning task: Robot
+    2019-05-21 17:29:29:        As user: test-wftmq9afc3ud@example.com
+    2019-05-21 17:29:29:         In org: 00Df0000003cuDx
+    2019-05-21 17:29:29:
+    ==============================================================================
+    MyObject
+    ==============================================================================
+    Test the MyObject listing page                                        | PASS |
+    ------------------------------------------------------------------------------
+    MyObject                                                              | PASS |
+    1 critical test, 1 passed, 0 failed
+    1 test total, 1 passed, 0 failed
+    ==============================================================================
+    Output:  /Users/boakley/dev/MyProject/robot/MyProject/results/output.xml
+    Log:     /Users/boakley/dev/MyProject/robot/MyProject/results/log.html
+    Report:  /Users/boakley/dev/MyProject/robot/MyProject/results/report.html
+
+Part 4: Creating a page object
+==============================
+
+Most projects are going to need to write custom keywords that are
+unique to that project. For example, NPSP has a keyword for filling in
+a batch gift entry form, EDA has a keyword with some custom logic for
+validating and affiliated contact, and so on.
+
+The best way to create and organize these keywords is to place them in
+page object libraries. These libraries contain normal Python classes
+and methods which have been decorated with the ``pageobjects``
+decorator provided by CumulusCI. By using page objects, you can write
+keywords that are unique to a given page, making them easier to find
```

Comment:
I feel like there must be more to them than being easier to find and manage. Can you explain a little bit more what the page object infrastructure _does_ at runtime? Is it just an easier way of loading keywords? Or do the keywords have some magic associated with them by the decorator?

Reply:
This is a good comment.  You are right that maybe I haven't fully explained what page objects are here. They are introduced in the main robotframework section of the documentation, so I'm not going to add anything here. What I did do, however, was add a link at the very top of this file to encourage people to read that section before diving into the tutorial.

FWIW, the main benefit really is to make keywords easier to find and manage. That, and provides a bit more of a structured environment for creating keywords so that our keyword libraries aren't quite the wild wild west that they are right now.

Thanks for the comment!